receiver/splunkhecreceiver/receiver.go

```diff
@@ -302,7 +302,7 @@ func (r *splunkReceiver) handleRawReq(resp http.ResponseWriter, req *http.Reques
 		defer r.gzipReaderPool.Put(reader)
 	}
 
-	resourceCustomizer := r.createResourceCustomizer(req)
+	ctx = r.contextWithAccessToken(ctx, req)
```

Comment:
we will need to place this behind a featuregate to make sure we don't introduce a breaking change. Please read about feature gates under https://github.com/open-telemetry/opentelemetry-collector/tree/main/featuregate#collector-feature-gates